drivers/dma/dma_stm32_v3.c

```diff
@@ -0,0 +1,763 @@
+
+static const uint32_t table_src_size[4] = {
+	LL_DMA_SRC_DATAWIDTH_BYTE,
+	LL_DMA_SRC_DATAWIDTH_HALFWORD,
+	LL_DMA_SRC_DATAWIDTH_WORD,
+	LL_DMA_SRC_DATAWIDTH_DOUBLEWORD,
+};
+
+static const uint32_t table_dst_size[4] = {
+	LL_DMA_DEST_DATAWIDTH_BYTE,
+	LL_DMA_DEST_DATAWIDTH_HALFWORD,
+	LL_DMA_DEST_DATAWIDTH_WORD,
+	LL_DMA_DEST_DATAWIDTH_DOUBLEWORD,
+};
+
+static const uint32_t table_priority[4] = {
+	LL_DMA_LOW_PRIORITY_LOW_WEIGHT,
+	LL_DMA_LOW_PRIORITY_MID_WEIGHT,
+	LL_DMA_LOW_PRIORITY_HIGH_WEIGHT,
+	LL_DMA_HIGH_PRIORITY,
+};
+
+uint32_t dma_stm32_id_to_channel(uint32_t id)
+{
+	static const uint32_t channel_nr[] = {
+		LL_DMA_CHANNEL_0,  LL_DMA_CHANNEL_1,  LL_DMA_CHANNEL_2,  LL_DMA_CHANNEL_3,
+		LL_DMA_CHANNEL_4,  LL_DMA_CHANNEL_5,  LL_DMA_CHANNEL_6,  LL_DMA_CHANNEL_7,
+		LL_DMA_CHANNEL_8,  LL_DMA_CHANNEL_9,  LL_DMA_CHANNEL_10, LL_DMA_CHANNEL_11,
+		LL_DMA_CHANNEL_12, LL_DMA_CHANNEL_13, LL_DMA_CHANNEL_14, LL_DMA_CHANNEL_15,
+	};
+
+	__ASSERT_NO_MSG(id < ARRAY_SIZE(channel_nr));
+
+	return channel_nr[id];
+}
+
+static int dma_stm32_get_src_inc_mode(enum dma_addr_adj increment, uint32_t *ll_increment)
+{
+	switch (increment) {
+	case DMA_ADDR_ADJ_INCREMENT:
+		*ll_increment = LL_DMA_SRC_INCREMENT;
+		break;
+	case DMA_ADDR_ADJ_NO_CHANGE:
+		*ll_increment = LL_DMA_SRC_FIXED;
+		break;
+	case DMA_ADDR_ADJ_DECREMENT:
+		LOG_ERR("Decrement mode not supported for source address adjustment");
+		return -ENOTSUP;
+	default:
+		LOG_ERR("Invalid source increment mode: %d", increment);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_stm32_get_dest_inc_mode(enum dma_addr_adj increment, uint32_t *ll_increment)
+{
+	switch (increment) {
+	case DMA_ADDR_ADJ_INCREMENT:
+		*ll_increment = LL_DMA_DEST_INCREMENT;
+		break;
+	case DMA_ADDR_ADJ_NO_CHANGE:
+		*ll_increment = LL_DMA_DEST_FIXED;
+		break;
+	case DMA_ADDR_ADJ_DECREMENT:
+		LOG_ERR("Decrement mode not supported for destination address adjustment");
+		return -ENOTSUP;
+	default:
+		LOG_ERR("Invalid destination increment mode: %d", increment);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_stm32_get_ll_direction(enum dma_channel_direction direction, uint32_t *ll_direction)
+{
+	switch (direction) {
+	case MEMORY_TO_MEMORY:
+		*ll_direction = LL_DMA_DIRECTION_MEMORY_TO_MEMORY;
+		break;
+	case MEMORY_TO_PERIPHERAL:
+		*ll_direction = LL_DMA_DIRECTION_MEMORY_TO_PERIPH;
+		break;
+	case PERIPHERAL_TO_MEMORY:
+		*ll_direction = LL_DMA_DIRECTION_PERIPH_TO_MEMORY;
+		break;
+	default:
+		LOG_ERR("Direction error. %d", direction);
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int dma_stm32_get_priority(uint8_t priority, uint32_t *ll_priority)
+{
+	if (priority > ARRAY_SIZE(table_priority)) {
+		LOG_ERR("Priority error.");
+		return -EINVAL;
+	}
+
+	*ll_priority = table_priority[priority];
+	return 0;
+}
+
+inline bool dma_stm32_is_tc_active(const DMA_TypeDef *dma, uint32_t channel)
+{
+	return LL_DMA_IsActiveFlag_TC(dma, channel);
+}
```

Comment:
1. Don't mark external functions as `inline`.
```suggestion
bool dma_stm32_is_tc_active(const DMA_TypeDef *dma, uint32_t channel)
{
	return LL_DMA_IsActiveFlag_TC(dma, channel);
}
```
Ditto all. (If you want the inlining, move these to the `.h` and make them `static inline` instead)

2. The LL functions should be used directly as these are mere wrappers

Reply:
I replaced all "useless" wrappers with direct LL function calls, avoiding the use of inline since LL functions already compile as such.